conjure-undertow-lib/src/main/java/com/palantir/conjure/java/undertow/lib/Endpoint.java

```diff
@@ -60,6 +63,11 @@
         return Optional.empty();
     }
 
+    /** Endpoint tags corresponding to the endpoint definition. */
+    default List<String> tags() {
```

Comment:
I chose List over Set here to make it clear that order is retained from the definition. In many cases I would expect values not to work as simple lookups e.g. `collection.contains(value)`, as there may be a known prefix with a dynamic suffix. Thoughts?

Reply:
hmm aren't the tags sets into the conjure definition? I think we should keep the uniqueness guarantee

Reply:
ArgumentDefinition tags are a list, EndpointDefinition tags are a set. imo lists are easier to deal with at runtime, and certainly more concise in memory. Lookups are faster in small lists than small sets.

Reply:
We provide the uniqueness constraint at the conjure schema level, moving the validation cost from execution time to build time.

Reply:
hmm ok we should consolidate on one at the IR level, thats my bad. I'm more concerned about the information the types give (that you should expect unique values) rather than the memory/lookup cost of lists vs. sets

Reply:
I've been thinking about these carrying composite data more than constant string values, where the set doesn't provide any help since we already need to handle deduplication by a common component. Either way the collection type shouldn't matter much since I expect it will only be used in setup, not on a per-request basis.

Reply:
hmm, i'm a bit concerned about the composite data. the original design of tags was string because we explicitly wanted to limit the complexity of the information in the tag.

Reply:
Are we comfortable adding more complex data directly to the IR?

Reply:
When we discussed the design, we concluded that the single string was sufficient because it could handle both cases. It's possible I missed a subsequent discussion.

Reply:
I think we would need to discuss alternatives like tags being `map<string, string>` or `map<string, any>`. unclear if we want that much complexity in the IR. Happy to hop on a call and discuss